src/utilities/api-requests.ts

```diff
@@ -10,13 +11,22 @@ import axios from "axios";
  */
 export async function request<T>(uri: string, defaultValue: T): Promise<T> {
     let data = defaultValue;
-    await axios
-        .get(uri)
-        .then((res) => {
-            data = res.data as T;
-        })
-        .catch((error) => {
-            console.error(`Request to ${uri} failed with\n${error}`);
-        });
+    let attempt = 0;
+
+    while (data == defaultValue && attempt < 5) {
+        await axios
+            .get(uri)
+            .then((res: AxiosResponse) => {
+                data = res.data as T;
+            })
+            .catch((reason: AxiosError) => {
+                console.error(`Request to ${uri} failed with\n${reason}`);
+            });
+
+        if (data == defaultValue) {
```

Comment:
Codacy found an issue: [Expected '===' and instead saw '=='.](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=5953345)